cli/command/registry/login.go

```diff
@@ -176,11 +171,7 @@ func loginWithStoredCredentials(ctx context.Context, dockerCLI command.Cli, auth
 		authConfig.IdentityToken = response.IdentityToken
 	}
 
-	if err := storeCredentials(dockerCLI.ConfigFile(), authConfig); err != nil {
-		return "", err
-	}
-
-	return response.Status, err
+	return storeCredentials(dockerCLI.ConfigFile(), authConfig)
```

Comment:
Ah! The existing code shadowed the error; so this function would return when failing to _store_ the credentials, but the error _returned_ is the error from earlier?? `return response.Status, err` returns the `err` from `response, err := dockerCLI.Client().RegistryLogin(ctx, authConfig)`, but even on failure, it would still try to store the credentials 🤔 


I suspect that was a bug!